### ISSUE TITLE
[4.0] Dashboard title icons

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#26217]
The icons used with the various dashboard titles are created from a language string :(

https://github.com/joomla/joomla-cms/blob/eaa4f48ab0821ed08c84bb570cc96bb6b0843439/administrator/components/com_cpanel/View/Cpanel/HtmlView.php#L62

https://github.com/joomla/joomla-cms/blob/eaa4f48ab0821ed08c84bb570cc96bb6b0843439/administrator/language/en-GB/en-GB.com_cpanel.ini#L13-L14

They really should not be as a css class is not a language string.

**Comment by infograf768** [comment]
Agree. See `$keyIcon = $prefix . $sectionkey . '_ICON';` and
`$icon = Text::_($keyIcon);`

**Comment by infograf768** [comment]
@chmst
As you introduced this in the new backend template, can you figure another way?
https://github.com/joomla/joomla-cms/commit/b04e5507b64e3ae1965ba60b599bf57cb3d4a269#diff-038a8497dd9ef101525de218e0877ae9

**Comment by chmst** [comment]
I agree that this is bad. But have no idea how to resolve this.  
The icon for the dashboard is not stored in the database, in this view we have the name of the component here, nothing else. 

@Hackwar can you help here?

**Comment by infograf768** [comment]
If I understand well, the only reason why it is not harcoded is because it was desired that 3pds be able to create custom Presets where the heading of the custom CPanel dashboard would fit a specific non-core extension.
Taking into account all the issues we have with this new feature and mod_submenu, I suggest this part of the new admin template to be refactored.

**Comment by infograf768** [comment]
I am quite confused by the code in this file.
In which cases can we get `$parts[1]` from `$parts = explode('.', $extension);` ?

After clarification, i.e. what kind of custom dashboard/presets (i.e. non-core ones) for which we need all that code instead of simply hardcoding $icon value depending on the default possible core dashboards, we could simplify the file and get rid of the lang strings for $icon.

**Comment by brianteeman** [comment]
As much as I dont like hard coding anything - with the current code base that's the only option I can see to handle this.

**Comment by infograf768** [comment]
Note:
2 strings missing if we do not change this code.
```
COM_CPANEL_DASHBOARD_PRIVACY_TITLE="Privacy Dashboard"
COM_CPANEL_DASHBOARD_PRIVACY_ICON="lock"
```

**Comment by infograf768** [comment]
> As much as I dont like hard coding anything - with the current code base that's the only option I can see to handle this.

The icon is always used for `ToolbarHelper::title`
So it is not as if we were hardcoding it specially for this case.
Example for articles manager where `$icon` is `stack article`:
`ToolbarHelper::title(Text::_('COM_CONTENT_ARTICLES_TITLE'), 'stack article');`

**Comment by infograf768** [comment]
Bugsquad discussion:
@Hackwar  is against any change, including setting a different behavior (hard coding $icon) for core.
The feature is supposed to cope with specific dashboards created by 3pd.

**Comment by brianteeman** [comment]
Then the code needs to be improved to be able to work without using a language string to set a class

**Comment by brianteeman** [comment]
Can this be used
https://github.com/joomla/joomla-cms/blob/1414814a34cffbba5d6365eeee5eff906d55a7cb/administrator/modules/mod_menu/Menu/CssMenu.php#L497

**Comment by infograf768** [comment]
I made a PR hardcoding icons as this is specific to Toolbarhelper::title()
PLT will decide.

See: https://github.com/joomla/joomla-cms/pull/26252

**Comment by ghost** [comment]
Closed as having Pull Request #26252
